web/controllers/user_controller.ex

```diff
@@ -34,6 +35,21 @@ defmodule CodeCorps.UserController do
     end
   end
 
+  def github_connect(conn, code) do
+    current_user = Guardian.Plug.current_resource(conn)
+    with {:ok, user} <- Github.connect(current_user, code)
+    do
+      # {:ok, user}
+      conn |> render(:show, data: user)
+    else
+      {:error, _error} ->
+        # {:error, error}
+        conn
+        |> put_status(404)
+        |> render(CodeCorps.ErrorView, "404.json-api")
+    end
```

Comment:
The else block here would be something like

```elixir
else
  {:error, %Ecto.Changeset{} = changeset} -> render_validation_server_error
  {:error, _error} -> render_internal_server_error
```

There should be examples for an internal server error somewhere. Try searching for "500". Alternatively, we could leave that case out, in which case, the `with` will raise an unmatched clause error, which will also cause the api to render a 500.